### PR TITLE
Gocast

### PR DESCRIPTION
**Pull request by stamp**

### CODE REVIEW

File: nodes/stampzilla-chromecast/state.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+func (s *State) Add(c *Chromecast) {
+	c.publish = s.Publish
+
+	if s.Devices == nil {
+		s.Devices = make(map[string]*Chromecast, 0)
+	}
+
+	s.Lock()
+	s.Devices[c.Name()] = c
```
Comment (jonaz) [inline nodes/stampzilla-chromecast/state.go]: Should use UUID here? Not name

File: nodes/stampzilla-chromecast/state.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	s.Unlock()
+
+	s.Publish()
+}
+
+func (s *State) Remove(c *Chromecast) {
+	_, ok := s.Devices[c.Name()]
```
Comment (jonaz) [inline nodes/stampzilla-chromecast/state.go]: Should use UUID here? Not name

File: nodes/stampzilla-chromecast/state.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+	if ok {
+		s.Lock()
+		delete(s.Devices, c.Name())
```
Comment (jonaz) [inline nodes/stampzilla-chromecast/state.go]: Should use UUID here? Not name

**Comment by jonaz** [comment]
Also getters in gocast/device.go No need to prefix them with Get....
This is enought: 
Ip()
Port()
Uuid()

File: nodes/stampzilla-chromecast/state.go (hunk 3 above)
Comment (stamp) [inline nodes/stampzilla-chromecast/state.go]: You are probably right.. as usual :)

**Comment by stamp** [comment]
I'm thinking about removing the getters, they actually dont do mutch

**Comment by jonaz** [comment]
Yes they matter VERY much. The properties are private and should not be accesses from the outside without getters. Making them public will make the setters unreliable since they can be modified without control from anywhere.

**Comment by jonaz** [comment]
Approved :)

**Pull request merged**